File: app/src/main/java/ae/innovativesolutions/wisdomhours/data/local/prefs/AppPreferencesHelper.java


package ae.innovativesolutions.wisdomhours.data.local.prefs;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

import ae.innovativesolutions.wisdomhours.di.scopes.PreferenceScope;

/**
 * Created by Zuhaib Ahmad on 07/07/17.
 */

public class AppPreferencesHelper implements PreferencesHelper {

	private final SharedPreferences mPrefs;

	@Inject
	public AppPreferencesHelper(Context context, @PreferenceScope String prefFileName) {
		mPrefs = context.getSharedPreferences(prefFileName, Context.MODE_PRIVATE);
	}
}

File: app/src/main/java/com/zuhaibahmad/template/data/local/prefs/AppPreferencesHelper.kt
package com.zuhaibahmad.template.data.local.prefs

import android.content.Context
import com.zuhaibahmad.template.di.scopes.PreferenceScope
import javax.inject.Inject

/**
 * Created by Zuhaib Ahmad on 07/07/17.
 */

class AppPreferencesHelper
@Inject constructor(
	context: Context,
	@PreferenceScope prefFileName: String
) : PreferencesHelper {

	private val mPrefs = context.getSharedPreferences(prefFileName, Context.MODE_PRIVATE)

	override var isUserSignedIn: Boolean
		get() = mPrefs.getBoolean("isUserSignedIn", false)
		set(value) = mPrefs.edit().putBoolean("isUserSignedIn", value).apply()
}
